### PR TITLE
@hapi/hapi method return type

### PR DESCRIPTION
**Pull request by qwelias**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
  - You use toolkit to return something, you do not return toolkit itself
    - [lifecycle](https://hapi.dev/api/?v=18.4.2#lifecycle-methods)
  - ~[context](https://hapi.dev/api/?v=20.0.3#-hcontext)~
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/hapi__hapi/index.d.ts [hunk 1]
```diff
@@ -4051,7 +4051,8 @@ export namespace Lifecycle {
         (stream.Stream) |
         (object | object[]) |
         symbol |
-        ResponseToolkit;
+        Auth |
+        ResponseObject;
```
Comment (saboya) [inline types/hapi__hapi/index.d.ts:4055]: Why remove `ResponseToolkit` here? Per documentation, it seems that some if its members are returned here:

https://github.com/hapijs/hapi/blob/master/API.md#lifecycle-methods

Where would the toolkit signals and toolkit methods come from here?

**Review by saboya — CHANGES_REQUESTED** [review]

File: types/hapi__hapi/index.d.ts (hunk 1 above)
Comment (qwelias) [reply types/hapi__hapi/index.d.ts:4055]: @saboya not really sure what you're talking about.

If you mean
> a toolkit signal:
h.abandon - abort processing the request.
h.close - abort processing the request and call end() to ensure the response is closed.
h.continue - continue processing the request lifecycle without changing the response.

Then those are all of `symbol` type.

If you mean
> a toolkit method response:
h.response() - wraps a plain response in a response object.
h.redirect() - wraps a plain response with a redirection directive.
h.authenticated() - indicate request authenticated successfully (auth scheme only).
h.unauthenticated() - indicate request failed to authenticate (auth scheme only).

Then those are either of `ResponseObject` or `Auth` type.

The toolkit itself is not the return type, you use the toolkit to return specific `symbol`s or `ResponseObject` or `Auth`.

And as you said
> Per documentation, it seems that some if its members are returned here:

None of it's members types includes the `ResponseToolkit` itself, you do not return the toolkit.

Comment (saboya) [reply types/hapi__hapi/index.d.ts:4055]: You're correct, that was a misinterpration on my part.

**Review by saboya — APPROVED** [review]

**Comment by qwelias** [comment]
@saboya I have dropped the commit with the change of a `h.context` type. Hapi docs are misleading -- while they say it's a symbol it could actually be anything

**Review by saboya — APPROVED** [review]

**Review by rbuckton — APPROVED** [review]

**Comment by qwelias** [comment]
Ready to merge

**Pull request merged**